javascript/ql/src/Security/CWE-020/IncompleteUrlSchemeCheck.ql

```diff
@@ -12,11 +12,46 @@
  */
 
 import javascript
-import semmle.javascript.dataflow.internal.AccessPaths
 
 /** A URL scheme that can be used to represent executable code. */
 class DangerousScheme extends string {
   DangerousScheme() { this = "data:" or this = "javascript:" or this = "vbscript:" }
+
+  /** Gets the name of this scheme without the `:`. */
+  string getWithoutColon() { this = result + ":" }
+
+  /** Gets the name of this scheme, with or without the `:`. */
+  string getWithOrWithoutColon() { result = this or result = getWithoutColon() }
+}
+
+/** Returns a node that refers to the scheme of `url`. */
+DataFlow::SourceNode schemeOf(DataFlow::Node url) {
+  // url.split(":")[0]
```

Comment:
do we want to support `url.split("/")[0]` as well?

Reply:
Done. [Updated query run](https://lgtm.com/query/2976842532241554709/) has no new results.

Reply:
On second thought, splitting by `/` will never work for detecting the `javascript:` scheme. I mean just look at the string `javascript:alert(1)`. No slashes. I'll revert this change again.